scripts/deploy/deploy.ts

```diff
@@ -48,48 +54,48 @@ export async function main(
 
     console.log("FeeController deployed to: ", feeController.address);
 
+    const PromissoryNoteFactory = await ethers.getContractFactory("PromissoryNote");
+    const borrowerNote = <PromissoryNote>await PromissoryNoteFactory.deploy("Arcade.xyz BorrowerNote", "aBN");
+    await borrowerNote.deployed()
+    const lenderNote = <PromissoryNote>await PromissoryNoteFactory.deploy("Arcade.xyz LenderNote", "aLN");
+    await lenderNote.deployed()
+    console.log("BorrowerNote deployed to:", borrowerNote.address);
+    console.log("LenderNote deployed to:", lenderNote.address);
+
     const LoanCoreFactory = await ethers.getContractFactory("LoanCore");
-    const loanCore = <LoanCore>await upgrades.deployProxy(LoanCoreFactory, [feeController.address], { kind: "uups" });
+    const loanCore = <LoanCore>await upgrades.deployProxy(LoanCoreFactory, [feeController.address, borrowerNote.address, lenderNote.address], { kind: "uups" });
     await loanCore.deployed();
-
     console.log("LoanCore deployed to:", loanCore.address);
 
-    const promissoryNoteFactory = await ethers.getContractFactory("PromissoryNote");
-    const borrowerNoteAddr = await loanCore.borrowerNote();
-    const borrowerNote = <PromissoryNote>await promissoryNoteFactory.attach(borrowerNoteAddr);
-    const lenderNoteAddr = await loanCore.lenderNote();
-    const lenderNote = <PromissoryNote>await promissoryNoteFactory.attach(lenderNoteAddr);
-
-    console.log("BorrowerNote deployed to:", borrowerNoteAddr);
-    console.log("LenderNote deployed to:", lenderNoteAddr);
+    // Grant correct permissions for promissory note
+    // Giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
```

Comment:
SideNote:
If we remove it from `deploy.ts`, we would need to add it to `bootstrap-state-with-loans.ts` because it calls `crreateLoanTermsSignature` via `bootstrap-tools`.

Reply:
It may make sense to keep it in `deploy.ts`, as `deploy.ts` is imported into `transfer-ownership` and into `bootstrap-state-with-loans`, in its current state they are both accessing it in a simple way.

Reply:
Furthermore, when moved to `setup-roles.ts `, `borrower`, `deployer` and `admin` signers require AccessControl permissions.

See this for example:

![Screen Shot 2022-05-24 at 8 13 53 PM](https://user-images.githubusercontent.com/42501101/170152114-e59ce3dd-5402-4124-a6f2-fbf85c06842b.png)

Reply:
It sounds like `bootstrap-state-with-loans` may need to run `setup-roles` as well as `deploy` then, right? If we don't set up the roles before it's not really a very useful script since it won't match real-world conditions.

Not sure what you mean by the last comment, only the `admin` should be able to call `initialize`. That admin would match the same account that ran the deploys cript and can call `grantRole` on all the other contracts.